### PR TITLE
wrote unit tests and fixed withdrawl method

### PR DESCRIPTION
**Pull request by whittyhandle**
I added some inline notes

### CODE REVIEW

File: CodingChallenge/src/Challenge79/Account.java [hunk 1]
```diff
@@ -16,18 +20,21 @@ public Account(String accountNum, double balance, String name, String email, Str
         this.phoneNum = phoneNum;
     }
 
-    public void deposit(double depositAmt){
-        this.balance = this.balance + depositAmt;
+    public double deposit(double depositAmt){ // added return type for testing
         System.out.println("You deposited: $" + depositAmt);
         System.out.println("Your new balance is: $" + this.balance);
+        return (this.balance = this.balance + depositAmt);
```
Comment (cvtc-Robert-Sanford) [inline CodingChallenge/src/Challenge79/Account.java:26]: By moving this.balance = this.balance + depositAmt to the return statement, does it still output correctly?

Comment (cvtc-Robert-Sanford) [reply CodingChallenge/src/Challenge79/Account.java:26]: I rewrote this method as well:

	public double deposit(double depositAmt) {
		String output = "";

		System.out.println(this.name + "\nYour current balance is: " + balance);

		if (depositAmt < 0) {
			System.out.println(invalidNum);
		} else {
			balance = balance + depositAmt;

			output = "You deposited: $" + depositAmt + "\nYour new balance is: $" + balance + "\n";

			System.out.println(output);
		}
		return balance;
	}

File: CodingChallenge/src/Challenge79/Account.java [hunk 2]
```diff
@@ -8,6 +8,10 @@
     private String email;
     private String phoneNum;
```
Comment (cvtc-Robert-Sanford) [inline CodingChallenge/src/Challenge79/Account.java:10]: Add a variable here:

	private String invalidNum = "Please enter a valid number\n";

**Review by cvtc-Robert-Sanford — COMMENTED** [review]
There was quite a bit that I changed/added.  they are not required but I noticed many of my errors and subsequently errors on Dave's iterations on my original code. Incremental improvements are a net plus.  Let me know what you think.

File: CodingChallenge/src/Challenge79/Account.java (hunk 1 above)
Comment (whittyhandle) [reply CodingChallenge/src/Challenge79/Account.java:26]: these look good to me

File: CodingChallenge/src/Challenge79/Account.java [hunk 3]
```diff
@@ -16,18 +20,21 @@ public Account(String accountNum, double balance, String name, String email, Str
...
-    public void deposit(double depositAmt){
-        this.balance = this.balance + depositAmt;
+    public double deposit(double depositAmt){ // added return type for testing
         System.out.println("You deposited: $" + depositAmt);
         System.out.println("Your new balance is: $" + this.balance);
+        return (this.balance = this.balance + depositAmt);
     }
 
-    public void withdrawal(double withdrawalAmt){
+    public double withdrawal(double withdrawalAmt){ // added return type and
```
[the remark this answers is not in the document]
Comment (whittyhandle) [reply CodingChallenge/src/Challenge79/Account.java:29]: yeah - I needed to do that for testing. I don't think we can test a method that returns void.

File: CodingChallenge/src/Challenge79/Account.java [hunk 4]
```diff
@@ -16,18 +20,21 @@ public Account(String accountNum, double balance, String name, String email, Str
...
         if(this.balance - withdrawalAmt < 0) {
             System.out.println("Insufficient account balance. Withdrawal denied.");
+            return this.balance;
         } else {
             System.out.println("You withdrew: $" + withdrawalAmt);
-            System.out.println("Your new balance is: $" + (this.balance - withdrawalAmt));
+            System.out.println("Your new balance is: $" + (this.balance - withdrawalAmt)); // I don't know if removes form the balance
+            this.balance = this.balance - withdrawalAmt;
+            return this.balance; // made it so it removes from the balance
         }
```
[the remark this answers is not in the document]
Comment (whittyhandle) [reply CodingChallenge/src/Challenge79/Account.java:39]: does `balance = balance - withdrawalAmt;` refer to `this.balance` is so we should be concomitantly using the this keyword.

**Pull request merged**